### DISCUSSION TITLE
Search results display are inconsistent

### DISCUSSION THREAD

**Discussion by jenniferpigeon** [microsoft-search/pnp-modern-search#2883] [Q&A]
Sometimes my search results return a preview in bold of the exact search terms I am looking for.  Other times, the search results do not show the search terms in the preview at all and instead you have to open the file and search within to find the search terms. 

Any idea why this would be happening?

**Comment by kasperbolarsen** [comment]
Are you asking about highlights, like in this: https://github.com/microsoft-search/pnp-modern-search/discussions/2746

**Reply by jenniferpigeon** [reply]
Yes, exactly.

**Reply by kasperbolarsen** [reply]
Does 2746 answer your question?

**Reply by jenniferpigeon** [reply]
No, because in a single search I get some results returned with highlights and other results without highlights. I wonder if the files that are not returning the highlighted results are somehow different and thereby not working?  All of my files are pdfs and all are searchable when inspected in Abode. I can’t seem to identify the reason for the inconsistency.

**Reply by kasperbolarsen** [reply]
I see. Lets try the basic first. In SharePoint search query tool or the SP Editor (Chrome extension) set the query to one of the PDFs that does not return any highlights when used in the web part.  If the result from the API does not contain any highlights then it is an API (or content) related issue and not a PnP modern search related issue.

**Reply by jenniferpigeon** [reply]
I downloaded the SharePoint Query Tool v. 2.10.0 and attempted to run it.  Under SharePoint Site URL, I tried several options: 1) the url of the pdf file itself, 2) the url of the document library containing the pdf and 3) the site url.  I also made sure to enter my credentials, entered my keyword into the Query text field but the status came back as HTTP/1.1 401 Unauthorized.  Under primary results I get "The query returned zero items!"  I even tried putting my tenant ID in during one attempt.  I left all other options as the default values.

Is there something in my SharePoint admin portal I need to adjust to allow this tool to run?

**Reply by wobba** [reply]
@jenniferpigeon this is the most common way to connect to SharePoint Online.
![image](https://user-images.githubusercontent.com/51104/229860581-0d886757-4b49-4582-9448-4625d54f05d4.png)

You can also use the site URL if you want like `https://contoso.sharepoint.com/sites/somesite`.

**Reply by jenniferpigeon** [reply]
Thanks for that - I got logged in.  I set my SharePoint Site url to my subsite containing the documents in question.  Not being familiar with this tool, can you tell me where and how to format my query to only search the PDF that does not return any highlights when used in the web part?  I assume I put input this into the Query Text box....I have tried several options with no luck.  

Also, will the results show up under the "Primary Results" tab? And how will the results be displayed in a way that I can determine if the text is highlighted?  Under HitHightlightedSummary?

**Reply by kasperbolarsen** [reply]
This is a good guide, https://www.youtube.com/live/BykEEMKNhHs?feature=share


I guess your query could be like path:https://contoso.sharepoint.com/sites/thesite/documents/thePDF.pdf*

**Reply by jenniferpigeon** [reply]
Great - I was able to narrow the query to that single pdf.  Now what to look for to determine if it contains the highlights?

**Reply by kasperbolarsen** [reply]
You should see the properties, like this
![image](https://user-images.githubusercontent.com/20593570/229939245-5d006c9e-26c2-4fcf-b73a-74f987223b69.png)

**Comment by jenniferpigeon** [comment]
The *HitHighlightedSummary* does not show the text that I am searching for
and expect to be highlighted. So does this mean it's an API issue?  Where
is the information from the *HitHighlightedSummary* field populated from?

**Reply by kasperbolarsen** [reply]
Just to rule out a root cause: does the HithighlighedSummary show up if you query one of the other PDFs?

**Comment by jenniferpigeon** [comment]
I queried one of the other PDFs in the SharePoint Search Query tool.  This PDF is one that DOES return highlights when searched from the web part.  In the HithighlighedSummary, it shows similar text to the first PDF queried, which does NOT include the highlighted text from the web part.

In summary:
PDF #1:  from the web part, I search the keywords "burden of proof" and it does NOT return those keywords in highlights.  Instead it returns the first couple of sentences from the PDF.
PDF #2: from the web part, I search the keywords "burden of proof" and it DOES return those keywords in highlights.  

What's strange is that the web part search results depend on the text searched.  For instance, if I search the text "licensing authority", PDF #1 and PDF #2 BOTH return highlights for that search without issue.  If I search "burden of proof", as in my example above, only PDF #2 returns highlights for that search.  From the SharePoint Search Query tool, both PDF #1 and PDF #2 have similar HitHighlightedSummary results, which happens to be the first couple of sentences from each PDF.

**Reply by wobba** [reply]
This is an API issue and nothing we can control via the web parts. You can try to log a support case with Microsoft.

**Comment by jenniferpigeon** [comment]
Unfortunately, Microsoft support was my first attempt to get this working and they directed me to the pnp-modern-search.

**Reply by wobba** [reply]
Tip, repro what you experience in classic search UX or modern search UX. Don't reference PnP web parts. Also, API issues on the SharePoint search API can be logged at https://github.com/SharePoint/sp-dev-docs/issues

**Comment by jenniferpigeon** [comment]
I cannot repro my steps because I am searching within a document library.
Microsoft support told me the following: "In site search and SPHome search
experiences, search results display a teaser or highlighted summary of
content relevant to the keyword search. This functionality, however, is not
available in list or library search. List or library search is a specific
feature that lacks this capability. A possible workaround is to implement
customization using modern PnP search web parts. Unfortunately,
out-of-the-box modern search web parts are not available in the service."

I am hopeful that PnP can help me since I have gotten it working to a
certain degree, but I just cannot find the logic behind why it works for
certain keywords and not others.

**Reply by wobba** [reply]
Document libraries will only show search in the library layout. If you go to site search or global search from e.g. SPHome, then a summary with highlight should be presented. If not, that would be a backend / API issue we cannot control in PnP web parts.

**Reply by jenniferpigeon** [reply]
Is the library layout within the PnP available layouts?  I do not see that.  Although, I am able to use the list and details list layout to get highlighted results, yet still inconsistent as described above.

**Reply by wobba** [reply]
A template purely using search results will never be the same as what you see in a list of library. To solve that you need to also fetch all items with more API calls to get the view, all column data etc. Certainly doable, but requires development.
